src/niMidi.cpp

```diff
@@ -234,21 +311,58 @@ class NiMidiSurface: public BaseSurface {
 			this->_sendSysex(CMD_TRACK_NAME, 0, numInBank, name);
 			// todo: level meters, volume, pan
 		}
-		// todo: navigate tracks, navigate banks
+		// todo: navigate tracks, navigate banks. NOTE: probably not here
+	}
+
+	void ClearSelected() {
+		// Clear all selected tracks. Copyright (c) 2010 and later Tim Payne (SWS)
```

Comment:
Another way to do this (which I use in OSARA) is this:

```
int Main_OnCommand(40297, 0); // Track: Unselect all tracks
// The master track has to be unselected separately.
GetSetMediaTrackInfo(GetMasterTrack(0), "I_SELECTED", &iSel);
```

I'm not sure if either has an advantage. This way will be undoable, but it's debatable as to whether that's a good thing or a bad thing.

Reply:
Yeah. Probably a matter of taste, I have no strong feelings about it. The Reaper actions are convenient but somehow a blackbox (admittedly so is the code behind the API calls). From a code readability perspective the direct API calls are somewhat better I guess...